linux/rsuite.sh

```diff
@@ -0,0 +1,183 @@
+
+EOF
+}
+
+view_apps() {
+    tmp=($(ls -d ../output/*/ 2> /dev/null | cut -f3 -d'/')) 
+    if (( ${#tmp[@]} == 0 )); then
+        echo "[Error] No disassembled apps found" >&2
+        exit 0
+    else
+        echo "The applications you've extracted are: "
+        for x in "${tmp[@]}"; do echo " " $x; done;
+        echo -e "\nTo build one of the apps use: $0 -b ${tmp[0]} -o newfile.apk"
+    fi
+}
+
+disassemble_app() {
+    [ ! -f $1 ] && echo "The file \"$1\" doesn't exist. Please specify a correct apk file." && exit 0
+    suggest_name=$(basename $1 | sed -e "s/.apk//g")
+    read -e -p "Provide a name to save this app: " -i "$suggest_name " response
+    if [ -d "$output_dir/$response" ]; then
+        opt_remove="N"
+        read -r -p "Directory exists. Can I remove it first? [y/N] " opt_remove
+        case "$opt_remove" in
+            [yY]) 
+                rm -r $output_dir/$response
+                java -jar "$BASEDIR/tools/others/apktool.jar" d "$1" -o "$output_dir/$response"
+                ;;
+            *)
+                read -e -p "Provide a name to save this app: " -i "$suggest_name " response
+                java -jar "$BASEDIR/tools/others/apktool.jar" d "$1" -f -o "$output_dir/$response" 
+                ;;
+        esac
+    else
+        java -jar "$BASEDIR/tools/others/apktool.jar" d "$1" -o "$output_dir/$response"
+    fi
+    
+    if [ -d "$output_dir/$response" ]; then
+        echo -e "\nExtracted files can be found here: \"$(readlink -e $output_dir/$response)\""
+    fi
+}
+
+decompile_app() {
+    [ ! -f $1 ] && echo "The file \"$1\" doesn't exist. Please specify a correct apk file." && exit 0
+    suggest_name=$(basename $1 | sed -e "s/.apk//g")
+    opt_remove="N"
+    echo "This option will remove any file with the name \"$suggest_name.jar\" (inside the ../output/ directory); will decompile the new file and open luyten."
+    read -r -p "Do you want to continue? [y/N] " opt
+    case "$opt" in
+        [yY]) 
+            rm $output_dir/$suggest_name.jar 2> /dev/null
+            bash "$BASEDIR/tools/dex2jar/d2j-dex2jar.sh" -f -o "$output_dir/$suggest_name.jar" "$1"
+            java -jar "$BASEDIR/tools/gui/luyten.jar" "$output_dir/$suggest_name.jar" &
+            ;;
+        *)
+            echo "[Info] Aborting.."
+            ;;
+    esac
+}
+build_app() {
+    [ ! -d $output_dir/$2 ] && echo -e "The application \"$2\" doesn't exist in the '../output/' directory.\nPlease use $0 -d <file.apk> first." && exit 0
+    if [ -f $3 ]; then
+        echo "The file \"$3\" already exist."
+        read -r -p "Do you want to replace it? [y/N] " opt
+        case "$opt" in
+            [yY]) 
+                rm $3
+                ;;
+            *)
+                echo -e "\n[Info] So.. choose a diferent output name.. :|"
+                exit 0
+                ;;
+        esac
+    fi
+
+    if [ -z $1 ]
+    then
+        echo "Optimazing the images in $output_dir/$2 - takes a while"
```

Comment:
This will take a while in fact. And it's not a requirement to build an application. It would be an awesome to add an extra parameter to skip this phase. If you find better to add an issue to implement this in future releases I'm fine with it.